flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/history/OnlyExecutionGraphJsonArchivist.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.webmonitor.history;
+
+import org.apache.flink.runtime.executiongraph.AccessExecutionGraph;
+import org.apache.flink.runtime.scheduler.ExecutionGraphInfo;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * Interface for all classes that want to participate in the archiving of job-related json responses
+ * but only provide {@link AccessExecutionGraph}-related information.
+ */
+public interface OnlyExecutionGraphJsonArchivist extends JsonArchivist {
```

Comment:
This fixes the `ExecutionGraphInfo` bad interface that was introduced in [FLINK-21188](https://issues.apache.org/jira/browse/FLINK-21188): `JsonArchivist` should not implement two methods as it will always rely. on `ExecutionGraphInfo`. This new interface `OnlyExecutionGraphJsonArchivist` is introduced to cover `AbstractAccessExecutionGraphHandler` implementation which don't care about `ExecutionGraphInfo`

Reply:
Why are we bothering with this distinction in the first place? Any archivist working on the `ExecutionGraphInfo` has access to the `AccessExecutionGraph` anyway. It should be trivial to adjust existing `JsonArchivist` implementations accordingly.

Reply:
The idea is to not bother about the `ExecutionGraphInfo`  in the implementations like the implementing classes do by deriving from `AbstractAccessExecutionGraphHandler`